specification/apicenter/data-plane/ApiCenter.DataApi/preview/2024-02-01-preview/azure.apicenter.dataapi.json

```diff
@@ -0,0 +1,1644 @@
+          },
+          "x-ms-identifiers": []
+        },
+        "customProperties": {
+          "type": "object",
+          "description": "The custom metadata defined for API entries.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "lastUpdated": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Last updated date and time."
+        }
+      },
+      "required": [
+        "name",
+        "title",
+        "kind",
+        "lastUpdated"
+      ]
+    },
+    "Models.ApiContact": {
+      "type": "object",
+      "description": "API contact information",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the contact.",
+          "maxLength": 100
+        },
+        "url": {
+          "type": "string",
+          "format": "uri",
+          "description": "URL for the contact."
+        },
+        "email": {
+          "type": "string",
+          "description": "Email address for the contact.",
+          "maxLength": 100
+        }
+      },
+      "required": [
+        "name",
+        "url",
+        "email"
+      ]
+    },
+    "Models.ApiDefinition": {
+      "type": "object",
+      "description": "API definition.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Definition identifier.",
+          "maxLength": 50,
+          "readOnly": true
+        },
+        "title": {
+          "type": "string",
+          "description": "The name of the API definition.",
+          "maxLength": 50
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the API definition.",
+          "maxLength": 1000
+        },
+        "specification": {
+          "$ref": "#/definitions/Models.ApiSpecification",
+          "description": "The API specification details."
+        }
+      },
+      "required": [
+        "name",
+        "title"
+      ]
+    },
+    "Models.ApiDeployment": {
+      "type": "object",
+      "description": "API Deployment resource model.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Deployment identifier.",
+          "maxLength": 50,
+          "readOnly": true
+        },
+        "title": {
+          "type": "string",
+          "description": "The name of the deployment.",
+          "maxLength": 50
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the deployment.",
+          "maxLength": 1000
+        },
+        "environment": {
+          "type": "string",
+          "description": "The name of the deployment environment."
+        },
+        "server": {
+          "$ref": "#/definitions/Models.ApiDeploymentServer",
+          "description": "The server information of the API deployment."
+        },
+        "customProperties": {
```

Comment:
The custom properties are not only strings, it might contain props of other types including complex ones. I suggest we take similar declaration as in [control plane spec](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/apicenter/resource-manager/Microsoft.ApiCenter/stable/2024-03-01/apicenter.json#L2796).